Userland/Utilities/useradd.cpp

```diff
@@ -157,16 +158,23 @@ int main(int argc, char** argv)
 
     if (putpwent(&p, pwfile) < 0) {
         perror("putpwent");
-        return 1;
+        goto fail;
```

Comment:
Please don't use `goto` unless absolutely necessary - a `ScopeGuard` would be much more idiomatic, and would also allow us to get rid of the `fclose()` calls before `return 0`.